internal/cba.go

```diff
@@ -97,8 +97,11 @@ func getTransportConfig(settings *DialSettings) (*transportConfig, error) {
 	}
 
 	// If client cert is found, use that over S2A.
+	// If DefaultMTLSEndpoint is not set, skip S2A.
 	// If MTLS is not enabled for the endpoint, skip S2A.
-	if clientCertSource != nil || !mtlsEndpointEnabledForS2A() {
+	// If custom HTTP client is provided, skip S2A.
+	if clientCertSource != nil || !mtlsEndpointEnabledForS2A() ||
+		settings.DefaultMTLSEndpoint == "" || settings.HTTPClient != nil {
```

Comment:
IMO this conditional is complicated enough that it should be wrapped by a function, maybe something like this:

```golang
func ShouldUseS2A($WHATEVER_PARAMS_YOU_NEED) bool
```

What do you think?